client/state/login/magic-login/actions.js

```diff
@@ -72,66 +66,48 @@ export const fetchMagicLoginRequestEmail = email => dispatch => {
 			type: MAGIC_LOGIN_SHOW_CHECK_YOUR_EMAIL_PAGE,
 			email
 		} );
-	} ).catch( error => {
+	} ).catch( ( error ) => {
 		dispatch( {
 			type: MAGIC_LOGIN_REQUEST_LOGIN_EMAIL_ERROR,
 			error: error.message,
 		} );
 	} );
 };
 
-const authError = error => {
-	return {
-		type: MAGIC_LOGIN_REQUEST_AUTH_ERROR,
-		error,
-	};
-};
-
-const authSuccess = () => {
-	return {
-		type: MAGIC_LOGIN_REQUEST_AUTH_SUCCESS,
-	};
-};
-
+// @TODO should this move to `/state/data-layer`..?
 export const fetchMagicLoginAuthenticate = ( email, token, tt ) => dispatch => {
 	dispatch( { type: MAGIC_LOGIN_REQUEST_AUTH_FETCH } );
 
-	const postData = {
-		client_id: config( 'wpcom_signup_id' ),
-		client_secret: config( 'wpcom_signup_key' ),
-		email,
-		token,
-		tt,
-	};
-
 	request
 		.post( AUTHENTICATE_URL )
 		.withCredentials()
-		.send( postData )
 		.set( {
 			Accept: 'application/json',
 			'Content-Type': 'application/x-www-form-urlencoded',
 		} )
-		.end( ( error, response ) => {
-			const { ok, statusCode } = response;
-
-			if ( error && ! response ) {
-				dispatch( authError( statusCode || 500 ) );
-				return;
-			}
-
-			if ( ok && statusCode === 200 ) {
-				dispatch( authSuccess() );
-				return;
-			}
-
-			dispatch( authError( statusCode || 403 ) );
+		.send( {
+			client_id: config( 'wpcom_signup_id' ),
+			client_secret: config( 'wpcom_signup_key' ),
+			email,
+			token,
+			tt,
+		} )
+		.then( ( response ) => {
+			dispatch( {
+				type: LOGIN_REQUEST_SUCCESS,
+				data: get( response, 'body.data' ),
+				// @TODO figure how we should treat `rememberMe`...
```

Comment:
Any opinions here? I'm not 💯 on the implications of setting `rememberMe` here -- does it just affect the TTL on the cookie?